---
output: 
  bookdown::pdf_book:
    base_format: rticles::elsevier_article
    keep_tex: true
    toc: false
    number_sections: false

title: "The effect of radiotherapy on the transcriptional profile of human cell lines"

abstract: |
  placeholder


author: 
- name: Stuart Lee
  email: lee.s@wehi.edu.au
  affiliation: Department of Econometrics and Business Statistics
- name: Lipi Shukla
  email: placeholder
  affiliation: placeholder
- name: Ramin Shayan
  email: placeholder
  affiliation: placeholder
- name: Matthew Ritchie
  email: mritchie@wehi.edu.au
  affiliation: Molecular Medicine Division, Walter and Eliza Hall Institute

# address:
#   - code: Some Institute of Technology
#     address: Department, Street, City, State, Zip
#   - code: Another University
#     address: Department, Street, City, State, Zip

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, 
                      message = FALSE, 
                      warning = FALSE,
                      fig.path = "./figures/",
                      cache.path = "./cache/", 
                      dpi = 300)
library(tidyverse)
library(limma)
library(edgeR)
```

```{r preprocess, cache=TRUE}
load("data/counts.rda")
# correct sample PERI1_10Gy counts, this sample was split accross
# three lanes, so we merge its counts
peri1_lane_split <- stringr::str_detect(colnames(counts$counts),  "PERI1_10GY")
total_peri1_counts <- rowSums(counts$counts[, peri1_lane_split])

# update counts matrix
counts_mat <- matrix(0, 
                     nrow = nrow(counts$counts),
                     ncol = sum(!peri1_lane_split) + 1)
counts_mat[, seq_len(sum(!peri1_lane_split))] <- counts$counts[, !peri1_lane_split]
counts_mat[, ncol(counts_mat)] <- total_peri1_counts

colnames(counts_mat) <- c(colnames(counts$counts)[!peri1_lane_split],
                          colnames(counts$counts)[peri1_lane_split][1L])

# filter genes based on cpm
gene_filter <- rowSums(edgeR::cpm(counts_mat) > 0.5) >= 2
counts_mat <- counts_mat[gene_filter, ]

# create colData
samples <- dplyr::data_frame(sample_names = colnames(counts_mat)) %>%
  dplyr::mutate(sample_meta = stringr::str_split(sample_names, "_"),
         celltype =  purrr::map_chr(sample_meta, ~ .x[1]) %>% 
           stringr::str_replace(., "1|2", ""),
         replicate = purrr::map_chr(sample_meta, ~ .x[1]) %>% 
           stringr::str_extract(., "1|2") %>% 
           as.integer(.),
         treatment = purrr::map_chr(sample_meta, ~.x[2]),
         lane = purrr::map_chr(sample_meta, ~.x[length(.x)]) %>%
           stringr::str_replace_all(., "L00|\\.sam", "") %>% 
           as.integer(.),
         library.size = colSums(counts_mat),
         norm.factors = 1,
         group = as.factor(paste0(celltype, "_", treatment))
  ) %>% 
  dplyr::select(-sample_meta)

# rowData
geneanno_update <- geneanno[gene_filter, ] %>% 
  dplyr::select(-EntrezID) 


elist <- edgeR::DGEList(counts = counts_mat,
                        samples = as.data.frame(samples),
                        group = samples$group,
                        genes =  as.data.frame(geneanno_update))

```

(ref:qc) **A** The variance explained by each dimension from the eigenvalues 
of the multidimensional scaling. **B** MDS plot coloured by cell-line. **C** MDS
plot of the LEC contrast. There is clear differential expression between treatment.
**D** Mean-variance trend in expression estimated by voom.

(ref:de) **A** The corrected mean-variance plot. **B** Common DE genes counts
between and within contrasts. 

# Specifications


+------------------------------------+-----------------------------------------+
| Organism/cell line/tissue          |  Human, Normal Dermal Fibroblasts,      | 
|                                    |  Normal Epidermal Keratinocytes,        |
|                                    |  Pericytes,  Microvascular              |
|                                    |  Blood Endothelial Cells,               |
|                                    |  Lymphatic Endothelial Cells and        |
|                                    |  Adipose Derived Stem Cell              |
+------------------------------------+-----------------------------------------+
| Sex                                |  N/A                                    |
+------------------------------------+-----------------------------------------+
| Data format                        | Unmapped fastq and summarised counts.   |
+------------------------------------+-----------------------------------------+
| Experimental factors               | RNA was obtained from each cell line    |
|                                    | under normal conditions and after being |          
|                                    | treated with a single dose of           | 
|                                    | radiotherapy. The adipose stem cells and|
|                                    | lymphatic endothelial cells were treated|
|                                    | with additional fractionated doses of   |
|                                    | radiotherapy.                           |
+------------------------------------+-----------------------------------------+
| Experimental features              |                                         |
+------------------------------------+-----------------------------------------+
| Consent                            | N/A                                     |
+------------------------------------+-----------------------------------------+
| Sample source location             | Melbourne, Australia                    |
+------------------------------------+-----------------------------------------+


# Direct link to deposited data

# Introduction


# Experimental Design, Materials and Methods

## Sample information

The effect radiotherapy has on gene expression was measured 
in 6 different human embyronic cell-lines consisting of 
normal dermal fibroblasts (NDF), normal epidermal keratinocytes (NEK), pericytes
(PC), microvascular blood endothelial cells (MEC), lymphatic endothelial cells 
(LEC) and adipose derived stem cells (ASC). Each cell-line was subjected to a 
single dose of radiotherapy (10Gy) or no radiotherapy (0Gy, control). 
For the ASC and LEC cell-lines, there was also a fractionated dose (5 x 2Gy 
over a 48 hour period). Two replicates of each cell-line/treatment combination 
were available.

## RNA-seq sample preparation and sequencing 
^[note: this is only for MEC need to check with Lipi the same protocol was used for each cell type.]

Standardized numbers of MEC were plated in cell culture flasks and irradiated 
once 80-90% confluence was achieved. RNA extraction was undertaken at 4 hours 
using the QIAGEN® RNEasy Plus Universal Kit as per manufacturers instructions. 
Samples were then tested for purity and quality control using the Nanodrop 
Spectrophotometer (Thermo Fischer Scientific) and stored at -80 °C, until 
further processing.  Extraction of RNA at a 4-hour time point post 
radiotherapy-injury was utilized.  Each sample underwent RNA sequencing 
(100 base pair single end) in the Illumina HiSeq machine at the Australian 
Genome Research Facility (AGRF) in Melbourne.

## Quality-control and data preprocessing

Sequences were mapped to the _hg19_ reference genome using the _Rsubread_
program with default settings and gene-level counts were obtained by 
the _featureCounts_ procedure using default settings. Transcripts were annotated
using the _org.Hs.eg.db_ package. Analysis of the resulting counts matrix
was performed using the _edgeR_ and _limma_ R/Bioconductor packages. First, 
counts per million (cpm) were computed for each gene to remove differences
caused by different library sizes. One sample, PERI1_10Gy, had its libraries 
split across three sequencing lanes, thus its cpm values were merged together.
Genes were retained for further analysis if they had a baseline expression
level of 0.5 cpm in at least 2 samples. Counts were normalised using the
TMM method.

```{r normalisation}
# perform TMM normalisation
elist <- calcNormFactors(elist, method = "TMM")

mds_obj <- plotMDS(elist, ndim = 10, plot = FALSE) 

mds_data <- left_join(mds_obj$cmdscale.out %>% 
  as.data.frame(.) %>% 
  rownames_to_column(., var = "sample_names"),
  samples)

mds_redo <- as.dist(mds_obj$distance.matrix) %>% 
  cmdscale(., k = 10, eig = TRUE) 

prop_variance <- data_frame(dim = seq_len(10),
                            variance_explained = mds_redo$eig[seq_len(10)] / sum(mds_redo$eig))


bar_plot <- ggplot(prop_variance, aes(x = dim, y = variance_explained)) +
  geom_bar(stat = "identity") + 
  annotate("text", x = 10, y = 0.375, size =  10, label = "A") +
  scale_x_continuous(breaks = seq_len(10), labels = seq_len(10)) +
  xlab("Dimension") + ylab("Proportion of variance explained")
  
mds_by_cellline <- ggplot(mds_data, 
                          aes(x = V1, y = V2, colour = celltype, label = group)) +
  geom_point(stroke = 0) +
  annotate("text", x = 4, y = 5.5, size = 10, label = "B") +
  ggrepel::geom_text_repel(size = 2) +
  xlab("Leading logFC dim 1") + ylab("Leading logFC dim 2") +
  guides(colour = FALSE) #+
  # theme(legend.justification = c(1,0), legend.position=c(1,0),
  #       legend.key.width = unit(0.1, "native"))

# show there's DE within contrasts
mds_lec <- left_join(plotMDS(elist[, grepl("LEC", elist$samples$group)], 
                   ndim = 2, plot = FALSE)$cmdscale.out %>%
  as.data.frame(.) %>% 
  rownames_to_column(., var = "sample_names"),
  samples)

mds_by_lec <- ggplot(mds_lec, 
                     aes(x = V1, y = V2, 
                         label = group)) +
  geom_point(stroke = 0) +
  ggrepel::geom_label_repel(size = 2) +
  annotate("text", x = 1.3, y = 0.8, size = 10, label = "C") +
  xlab("Leading logFC dim 1") + ylab("Leading logFC dim 2") +
  theme(legend.justification = c(1,0), legend.position=c(1,0))
  

# mds_by_replicate <- ggplot(mds_data, 
#                            aes(x = V1, y = V3, 
#                                colour = as.factor(replicate))) + 
#   geom_point() +
#   guides(colour = guide_legend(title = "Replicate number")) +
#   xlab("Leading logFC dim 1") + ylab("Leading logFC dim 3") +
#   theme(legend.position = "bottom")
```

```{r voom}
design <- model.matrix(~0 + samples$group) 
colnames(design) <- unique(samples$group)

contr <-  makeContrasts(ASC_10GY-ASC_0GY, 
                        ASC_2GYx5-ASC_0GY, 
                        HMEC_10GY-HMEC_0GY, 
                        LEC_10GY-LEC_0GY, 
                        LEC_2GYx5-LEC_0GY, 
                        NHDF_10GY-NHDF_0GY,
                        NHEK_10GY-NHEK_0GY, 
                        PERI_10GY-PERI_0GY,
                        PERI_0GY-ASC_0GY, 
                        PERI_10GY-ASC_10GY,
                        LEC_0GY-HMEC_0GY, 
                        LEC_10GY-HMEC_10GY, levels=colnames(design))
colnames(contr) <- gsub(" ", "", colnames(contr))

block <- model.matrix(~ 0 + as.factor(samples$replicate))
colnames(block) <- c("rep1", "rep2")

vfit <- voomWithQualityWeights(elist, 
                               plot = FALSE, 
                               design = design,
                               save.plot = TRUE)

mean_variance_df <- data_frame(mean_signal = vfit$voom.xy$x,
                                 variance_signal = vfit$voom.xy$y,
                                 loess_line_x = vfit$voom.line$x,
                                 loess_line_y = vfit$voom.line$y)

voom_plot <- ggplot(mean_variance_df, aes(x = mean_signal, y = variance_signal)) +
  geom_point(stroke = 0) +
  geom_line(aes(x = loess_line_x, y = loess_line_y), colour = "blue") +
  annotate("text", x = 16, y = 1.6, size = 10, label = "D") +
  xlab("log2(count size + 0.5)") + ylab("sqrt(standard deviation)")
```

```{r qc-figure, fig.cap="(ref:qc)", out.height = "14cm", fig.align="center", fig.asp=0.8}

fig_cap <- "A The variance explained by each dimension from the eigenvalues of the multidimensional scaling. B MDS plot coloured by cell-line. C MDS plot coloured by treatment. D Mean-variance trend in expression."
gridExtra::grid.arrange(bar_plot, 
                        mds_by_cellline, 
                        mds_by_lec,
                        voom_plot, nrow = 2, ncol = 2)
```

During exploratory data analysis we performed multidimensional scaling (MDS) on
the expression matrix with cpm values. The first and second dimension  explain 
`r 100*round(prop_variance[['variance_explained']][1], 2)` per cent and
`r 100*round(prop_variance[['variance_explained']][2], 2)` of the 
variation in the data, respectively (figure \@ref(fig:qc-figure)). 
There is a clear separation between the 
different cell-lines and the replicate number from the MDS plots. We model
the heteroscedasticty in gene counts using the _voomWithQualityWeights_ procedure 
using  the cell-line specific radiotherapy treatment as a main effect 
(figure \@ref(fig:qc-figure)).
This results in down-weighting genes with systematically higher variation.
Furthermore, given the clear difference between replicate samples observed 
in the MDS plot, the _voomWithQualityWeights_ 
procedure also down-weights samples that are more variable compared to others.


## Differential expression analysis
```{r deg}
lm_fit <- lmFit(vfit)
lm_fit <- contrasts.fit(lm_fit, contrasts = contr)
lm_fit <- eBayes(lm_fit)

corrected_variance_plot <- data_frame(mean_signal = lm_fit$Amean,
                               variance_signal = sqrt(lm_fit$sigma),
                               trend = sqrt(lm_fit$s2.prior)) %>% 
  ggplot(., aes(x = mean_signal, y = variance_signal)) +
  geom_point() +
  geom_line(aes(x = mean_signal, y = trend), colour = "blue") +
  xlab("Average log-expression") +
  ylab("sqrt(sigma)")

dt_all <- decideTests(lm_fit, method = "global", p.value = 0.05 )
dt_by_contrast <- decideTests(lm_fit, method = "separate", p.value = 0.05)


# compute whether gene is sig up/down in a given contrast
upset_data <- as.data.frame(dt_by_contrast != 0) %>% 
  mutate_if(is.logical, as.integer)


top_tables <- lapply(seq_len(ncol(lm_fit)), 
                     function(i) 
                       as_data_frame(topTable(lm_fit, 
                                              coef = i, 
                                              n = nrow(lm_fit))) %>%
                       mutate(contrast = colnames(lm_fit)[i])) %>% 
  bind_rows() %>%
  group_by(contrast)
  
                                                
                                                
```



```{r de-figs, out.height = "6cm", fig.cap="(ref:de)"}
fig_cap <- "A The corrected mean-variance plot. B Common DE genes counts between and within contrasts."
corrected_variance_plot
# UpSetR::upset(upset_data,
#               nsets = ncol(upset_data),
#               nintersects = NA,
#               order.by = "freq")
```

After modelling the heteroscedasticty in the counts matrix, we used empirical
Bayes moderation to compute more precise estimates of gene-wise variablilty
and remove the dependence between the variance and mean expression level (figure \@ref(fig:de-figs).
Differential expression between all contrasts were assessed using moderated
t-statistics using a false-discovery rate (FDR) of 5 per cent. Using this criteria,
there were `r sum(rowSums(dt_by_contrast != 0) > 1)` differentially expressed genes 
in at least one contrast (figure \@ref(fig:de-figs)).

# Results


# Discussion







